### PR TITLE
Add new instance types

### PR DESCRIPTION
**Pull request by uthark**
Fixes the following error:

```
tflint -c ../.tflint.hcl
1 issue(s) found:

aks-cluster.tf:133:26: Error - "Standard_NC24ads_A100_v4" is an invalid value as vm_size (azurerm_kubernetes_cluster_node_pool_invalid_vm_size)
```

(sorted list in the first commit alphabetically, 2nd commit adds missing VM sizes)

### CODE REVIEW

**Comment by uthark** [comment]
@wata727 please take a look

File: rules/utils.go [hunk 1]
```diff
@@ -38,12 +38,16 @@ var validMachineSizes = []string{
 	"Standard_D1",
 	"Standard_D11",
 	"Standard_D11_v2",
+	"Standard_D11_v2_Promo",
```
Comment (wata727) [inline rules/utils.go:41]: The Promo series certainly seems to exist, but it doesn't seem to be mentioned anywhere in the official documentation. Where are these sources?
https://learn.microsoft.com/en-us/azure/virtual-machines/sizes

File: rules/utils.go [hunk 2]
```diff
@@ -427,42 +572,22 @@ var validMachineSizes = []string{
...
-	"Standard_GS5-16",
-	"Standard_GS5-16",
-	"Standard_GS5-8",
-	"Standard_GS5-8",
-	"Standard_H16",
-	"Standard_H16m",
-	"Standard_H16mr",
-	"Standard_H16r",
-	"Standard_H8",
-	"Standard_H8m",
```
Comment (wata727) [inline rules/utils.go]: Why are these removed?

Comment (uthark) [reply rules/utils.go]: 1. they are not returned by the API.
2. They are retired, see https://learn.microsoft.com/en-us/azure/cloud-services/cloud-services-sizes-specs#h-series

> On August 31, 2022, we're retiring the following H-series Azure VM sizes:
> 
> H8
> H8m
> H16
> H16r
> H16m
> H16mr
> H8 Promo
> H8m Promo
> H16 Promo
> H16r Promo
> H16m Promo
> H16mr Promo

Would you suggest to keep them?  I can update the PR.

Comment (uthark) [reply rules/utils.go]: @wata727 ping

Comment (wata727) [reply rules/utils.go]: Thank you for your explanation.
I was concerned if anyone was still using these sizes, but according to the article, it seems that no such user exists anymore.
